### ISSUE TITLE
Semicolons in Front Matter cause things to break when inside HTML Elements

### ISSUE THREAD

**Issue by polds** [spf13/hugo#784]
Ran into an issue where putting a semicolon in the Front Matter causes Hugo to have issues rendering HTML. I have only tested this in TOML and have not verified if it happens using JSON or YAML.

Please see the following snippet on how to reproduce:
 
**md file**

    +++
    style = "width: 60px;"

    [navigation]
    [[navigation.item]]
        style = "width: 60px;"
    +++

**html template**

    <a style="{{ .Params.style }}"></a>
    {{ range $item := .Params.navigation }}
	<ul>
	{{ range $item }}
		{{ with .style }}
			<li style="{{ . | safeHtml }}">{{ . | safeHtml }}</li>
		{{ end }}
	{{ end }}
	</ul>
    {{ end }}

With that HTML template if you view the source of the rendered page you will see the following:

    <a style="ZgotmplZ"></a>
	<ul>
		<li style="ZgotmplZ">width: 60px;</li>
	</ul>

As soon as you remove the semicolon in the key it goes away and you properly see `style="width: 60px"`.

I have done the following tests:

`<li style="{{ . | safeHtml }}">` - Buggy
`<li {{ . | safeHtml }}>` - Buggy
`<li style="{{ .style }}">` (removing the `{{ with .style }}`) - Buggy
`<{{ . | safeHtml }}>` - Works as intended

It seems to be affecting HTML elements directly.

**Comment by tatsushid** [comment]
I think this is caused by the Go official `html/template` package's security design described at http://golang.org/pkg/html/template/. The similar case is discussed at #347.

**Comment by anthonyfok** [comment]
Just a quick note that a solution has been found by the use of template.CSS (found in Go html/template documentation and suggested by a post on Stack Overflow) and a new "safeCss" action.  For example, when the relevant code has been added to Hugo, we may replace

    <li style="{{ . }}">{{ . }}</li>

with

    <li style="{{ . | safeCss }}">{{ . }}</li>

and `width: 60px;` with the semicolon will appear correctly.

Still working on the documentation.  Will commit tonight or tomorrow.  :-)

**Comment by tatsushid** [comment]
@anthonyfok I wonder how we can use `template.CSS` in a template because it's Go's type and now it isn't provided a way to define variable type at assigning value in a template. To use it, it should be done at parsing and setting frontmatter values. I don't say it can't but it would be difficult.

**Comment by tatsushid** [comment]
@anthonyfok Sorry, I misunderstood. As you say, if we introduce a new function `safeCSS` which takes a string argument and returns a `template.CSS` type value, it would work. I'll test it in my env. Thanks for your deep insight.

**Comment by tatsushid** [comment]
I tested it by simply implementing the `safeCSS` template function and confirmed it works. Do you already have the implementation in your source? If not, after adding test codes, I'll send PR.

**Comment by anthonyfok** [comment]
@tatsushid, thank you for the PR!

While I did have the implementation in my source on my laptop—it is only three lines of code per function that I could copy from `safeHtml` after all, haha—I did not even think of adding test codes, and wouldn't know where to start, so your contribution is timely and educational indeed!  :-)  Will merge (cherry-pick) your PR soon.

**Comment by anthonyfok** [comment]
Hi @tatsushid,

Sorry, I made yet another incompatible change to your wonderful contribution.
In order to follow the naming tradition of `safeHtml`, I have decided to rename your `safeCSS` to `safeCss`.  I have thought of adding an alias to the `funcMap`, but decided not to do it at this point unless there is a consensus to provide aliases like `safeHTML` and `safeURL` too.

Hope I didn't cause too much trouble!  Thank you for your understanding!
